src/setup-ganache.ts

```diff
@@ -31,42 +31,55 @@ export type NodeOptions = {
 };
 
 export default async function (): Promise<string> {
-  const server = fork(path.join(__dirname, 'ganache-server'), [], {
-    // Prevent the child process from also being started in inspect mode, which
-    // would cause issues due to parent and child sharing the port.
-    // See https://github.com/OpenZeppelin/openzeppelin-test-environment/pull/23
-    execArgv: process.execArgv.filter((opt) => opt !== '--inspect'),
-  });
+  if (!config.coverage) {
+    const server = fork(path.join(__dirname, 'ganache-server'), [], {
+      // Prevent the child process from also being started in inspect mode, which
+      // would cause issues due to parent and child sharing the port.
+      // See https://github.com/OpenZeppelin/openzeppelin-test-environment/pull/23
+      execArgv: process.execArgv.filter((opt) => opt !== '--inspect'),
+    });
 
-  const gasPrice = typeof config.node.gasPrice === 'string' ? config.node.gasPrice : undefined;
+    const gasPrice = typeof config.node.gasPrice === 'string' ? config.node.gasPrice : undefined;
 
-  const options: NodeOptions = {
-    ...config.node,
-    gasPrice,
-    accounts: accountsConfig,
-    coverage: config.coverage,
-  };
-  server.send(options);
+    const options: NodeOptions = {
+      ...config.node,
+      gasPrice,
+      accounts: accountsConfig,
+      coverage: config.coverage,
+    };
+    server.send(options);
 
-  const messageReceived: Promise<Message> = new Promise(
-    (resolve): ChildProcess => {
-      return server.once('message', resolve);
-    },
-  );
+    const messageReceived: Promise<Message> = new Promise(
+      (resolve): ChildProcess => {
+        return server.once('message', resolve);
+      },
+    );
 
-  const message = await messageReceived;
+    const message = await messageReceived;
 
-  switch (message.type) {
-    case 'ready':
-      if (server.channel) {
-        server.channel.unref();
-      }
-      server.unref();
-      return `http://localhost:${message.port}`;
-    case 'error':
-      server.kill();
-      throw new Error('Unhandled server error');
-    default:
-      throw new Error(`Uknown server message: '${message}'`);
+    switch (message.type) {
+      case 'ready':
+        if (server.channel) {
+          server.channel.unref();
+        }
+        server.unref();
+        return `http://localhost:${message.port}`;
+      case 'error':
+        server.kill();
+        throw new Error('Unhandled server error');
+      default:
+        throw new Error(`Uknown server message: '${message}'`);
+    }
+  } else {
+    if (process.send === undefined) {
+      throw new Error('Module must be started through child_process.fork for solidity-coverage support.');
```

Comment:
I think relying on this is rather fragile. It will probably not work with any parallel test runner, including Mocha 8. (Would parallel tests work with coverage anyway?)

A more robust way to do it would be to manually open a communication endpoint (rather than relying on the automatic one opened by Node.js in `fork`). We can improve this later in a backwards compatible way though.

Reply:
It would certainly not work with parallel tests. It is theoretically possible to support solidity-coverage with parallel tests but out of scope in this PR.